build_tools/print_driver_gpu_info.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python3
+"""
+Sanity check script for CI runners.
+
+Prints a small set of driver / GPU information from:
+  - amd-smi
+  - rocminfo
+
+It prefers binaries installed under ./build/bin (as produced by
+install_rocm_from_artifacts.py), and falls back to the PATH.
+
+Usage:
+    ./build_tools/print_driver_gpu_info.py
+"""
+
+import argparse
+import json
+import logging
+import re
+import subprocess
+import sys
+from typing import Dict, Tuple, Optional, List, Any
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+log = logging.info
+
+
+def run_cmd(cmd: list[str]) -> Tuple[bool, str]:
+    """
+    Run a command and return (ok, combined_output).
+
+    ok = True  -> command executed and returned success (exit code 0)
+    ok = False -> command missing or returned a non-zero exit code
+    """
+    try:
+        proc = subprocess.run(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True,
+            check=False,
+        )
+        ok = proc.returncode == 0
+        return ok, proc.stdout
+    except FileNotFoundError:
+        return False, f"{cmd[0]}: command not found"
+
+
+def run_tool_with_candidates(
+    tool_name: str, candidates: list[str]
+) -> Tuple[Optional[str], bool, str]:
+    """
+    Try running a tool from a list of candidate paths / names.
+
+    Returns (resolved_candidate, ok, output).
+
+    - resolved_candidate: the candidate string that was used, or None
+    - ok: True if the command ran and returned exit code 0
+    - output: combined stdout/stderr from the last attempt
+    """
+    last_out = ""
+    for cand in candidates:
+        ok, out = run_cmd([cand])
+        last_out = out
+        # If the tool is literally not found, try the next candidate
+        if "command not found" in out:
+            continue
+        # We found *something* (even if it failed with non-zero rc)
+        return cand, ok, out
+
+    # Nothing worked
+    return None, False, last_out or f"{tool_name}: command not found"
+
+
+def parse_amd_smi(output: str) -> Dict[str, Any]:
+    """
+    Parse a few useful fields from amd-smi / amdsmi text output.
+
+    Supports both the older AMD-SMI header and the newer AMDSMI Tool header.
+    """
+    info: Dict[str, Any] = {}
+
+    # New-style amdsmi header, e.g.:
+    # AMDSMI Tool: 26.1.0+65125c66 | AMDSMI Library version: 26.1.0
+    # | ROCm version: 7.11.0 | amdgpu version: 6.12.12 | ...
+    new_header = re.search(
+        r"AMDSMI Tool:\s*([^\|]+?)\s*\|"
+        r"\s*AMDSMI Library version:\s*([^\|]+?)\s*\|"
+        r"\s*ROCm version:\s*([^\|]+?)\s*\|"
+        r"\s*amdgpu version:\s*([^\|]+?)\s*\|",
+        output,
+    )
```

Comment:
I don't understand this code. Why not just run `amdsmi` and log the full output?

* This regex searching through command output is brittle as it depends on the exact format of the output from a tool.
* We don't know what information will be useful. Better to just log everything - that will also solve "Since VRAM data is not included in this PR, please create a GitHub Issue to track adding it in the future."